containers.go

```diff
@@ -338,17 +343,22 @@ func (r *containerStore) Save() error {
 	return r.lockfile.Touch()
 }
 
-func newContainerStore(dir string) (rwContainerStore, error) {
+func newContainerStore(dir string, rundir string, transient bool) (rwContainerStore, error) {
 	if err := os.MkdirAll(dir, 0700); err != nil {
 		return nil, err
 	}
+	if err := os.MkdirAll(rundir, 0700); err != nil {
+		return nil, err
+	}
 	lockfile, err := GetLockfile(filepath.Join(dir, "containers.lock"))
```

Comment:
These objects are not quite traditional locks — https://github.com/containers/storage/blob/74e37433a2a04ef9ba62d7ae15c9110933b9aca6/pkg/lockfile/lockfile_unix.go#L246 happens on every update of the protected JSON file.

Reply:
I don’t understand c/storage enough to be able to answer.

At least for the “additional layer/image stores”, we need them in exactly the same location as the JSON (and the layer data), so that they can be shared over the network.

I’m not sure about the container data.

Reply:
Seems good to have the json and lock file in the same directory if its e.g. used as additional store then, so I moved it to tmpfs only in the transient case.